packages/block-library/src/page-list/index.php

```diff
@@ -9,51 +9,91 @@
  * Build an array with CSS classes and inline styles defining the colors
  * which will be applied to the pages markup in the front-end when it is a descendant of navigation.
  *
- * @param  array $context Navigation block context.
+ * @param  array $attributes Block attributes.
+ * @param  array $context    Navigation block context.
  * @return array Colors CSS classes and inline styles.
  */
-function block_core_page_list_build_css_colors( $context ) {
+function block_core_page_list_build_css_colors( $attributes, $context ) {
 	$colors = array(
-		'css_classes'   => array(),
-		'inline_styles' => '',
+		'css_classes'           => array(),
+		'inline_styles'         => '',
+		'overlay_css_classes'   => array(),
+		'overlay_inline_styles' => '',
 	);
 
 	// Text color.
-	$has_named_text_color  = array_key_exists( 'textColor', $context );
+	$has_named_text_color  = array_key_exists( 'textColor', $attributes );
+	$has_picked_text_color = array_key_exists( 'customTextColor', $attributes );
 	$has_custom_text_color = isset( $context['style']['color']['text'] );
 
 	// If has text color.
-	if ( $has_custom_text_color || $has_named_text_color ) {
+	if ( $has_custom_text_color || $has_picked_text_color || $has_named_text_color ) {
 		// Add has-text-color class.
 		$colors['css_classes'][] = 'has-text-color';
 	}
 
 	if ( $has_named_text_color ) {
 		// Add the color class.
-		$colors['css_classes'][] = sprintf( 'has-%s-color', $context['textColor'] );
+		$colors['css_classes'][] = sprintf( 'has-%s-color', $attributes['textColor'] );
```

Comment:
Same as https://github.com/WordPress/gutenberg/pull/31149/files?file-filters%5B%5D=.js&file-filters%5B%5D=.json#r657796626 There's the `gutenberg_experimental_to_kebab_case` function that you can use here.

Same for background and the overlay colors below.